release/models/system/openconfig-system.yang

```diff
@@ -341,6 +347,12 @@ module openconfig-system {
           restarted.  The value is the timestamp in nanoseconds relative
           to the Unix Epoch (Jan 1, 1970 00:00:00 UTC).";
     }
+
+    leaf software-version {
+        type string;
+        description
+          "Operating system version of the primary routing engine.";
```

Comment:
i think this should not be the "primary" routing engine, but maybe "the currently active controller of the device" since this is more generic. if this were to describe a linux device it would be whatever device is currently responding to the gNMI agent afaict.

note that there's some discussion to have here that i'd like to prompt: this represents a simplification of the user querying this data, at the cost of denormalising -- today one needs to walk the set of components to find the OS that corresponds to the currently active routing engine, which requires more query logic. we should consider as a wider group whether this is the right trade-off to be making.

Reply:
I've added this to the openconfig operators group agenda for comment there.  Also appreciate public comments as Rob indicated.

Reply:
Thanks for reviewing this.  I am suggesting adding this as a second option for describing the software version, which is higher level / abstracted away from the components, so one does not have to be component-specific if they want to know this information.

Reply:
Why is just software-version being requested here, is there a use case?  If it makes sense to include software-version wouldn't you also want os-name and/or manufacturer unless that is expected to be part of the software-version string?

Reply:
#675 covers some of the discussion here. As discussed, this is denormalising the models -- since one should be able to get this from the components that are of type `OPERATING_SYSTEM` within `/components/component`.

The challenge from a consumer perspective is that having this stored there means you need to walk the list of the components to find the component, and then subsequently read the version from it. This would have the additional information.

The question for the group here is whether we feel that the use case of "what version of the OS is this device running" justifies the denormalisation of the model.

Reply:
OC Community has a suggestion to require this leaf value should be the exact same as component type=`OPERATING_SYSTEM`.

Reply:
I have updated the description to include this requirement, as well as the phrase "currently active controller of the device".

Reply:
Thanks! Looks like you also need to fast forward / resync your fork to master to resolve a conflict.

Reply:
I fast forwarded, and merged my three commits into one.